src/views/VideoView/VideoView.tsx

```diff
@@ -32,15 +32,23 @@ const VideoView: React.FC<RouteComponentProps> = () => {
   })
   const [addVideoView] = useMutation<AddVideoView, AddVideoViewVariables>(ADD_VIDEO_VIEW)
   const { state, updateWatchedVideos } = usePersonalData()
+  const timestampFromQuery = Number(useRouterQuery('time'))
 
   const [startTimestamp, setStartTimestamp] = useState<number>()
   useEffect(() => {
     if (startTimestamp != null) {
       return
     }
     const currentVideo = state.watchedVideos.find((v) => v.id === data?.video?.id)
-    setStartTimestamp(currentVideo?.__typename === 'INTERRUPTED' ? currentVideo.timestamp : 0)
-  }, [data?.video?.id, state.watchedVideos, startTimestamp])
+    const duration = data?.video?.duration ? Number(data?.video?.duration) : 0
+
+    if (timestampFromQuery && timestampFromQuery < duration) {
```

Comment:
if `timestampFromQuery < duration` is false, we still need to do something here. If we just don't call `setStartTimestamp`, the initial condition in this `useEffect` will never fail so we will just keep calling this whenever the `state.watchedVideos` changes